kolibri/plugins/learn/assets/src/modules/examViewer/handlers.js

```diff
@@ -76,16 +96,16 @@ export function showExam(store, params, alreadyOnQuiz) {
                   contentNodeMap[node.id] = node;
                 }
 
-                for (const question of questions) {
+                for (const question of allQuestions) {
                   question.missing = !contentNodeMap[question.exercise_id];
                 }
-
-                store.commit('examViewer/SET_STATE', {
-                  contentNodeMap,
-                  exam,
-                  questionNumber,
-                  questions,
-                });
+                (exam.question_sources = question_sources),
+                  store.commit('examViewer/SET_STATE', {
```

Comment:
These two lines seem odd like this -- should this instead be two separate lines like:

```
exam.question_sources = question_sources;
store.commit('examViewer/SET_STATE', {
```